cider/tests/functionality/DateTypeQueryTest.cpp

```diff
@@ -72,39 +80,45 @@ class DateRandomAndNullQueryOf2009Test : public CiderTestBase {
   DateRandomAndNullQueryOf2009Test() {
     table_name_ = "test";
     create_ddl_ = "CREATE TABLE test(col_a DATE, col_b DATE);";
-    input_ = {std::make_shared<CiderBatch>(QueryDataGenerator::generateBatchByTypes(
+    QueryArrowDataGenerator::generateBatchByTypes(
+        schema_,
+        array_,
         500,
         {"col_a", "col_b"},
         {CREATE_SUBSTRAIT_TYPE(Date), CREATE_SUBSTRAIT_TYPE(Date)},
         {2, 3},
         GeneratePattern::Random,
         14245,
-        14609))};
+        14609);
   }
 };
 
 TEST_F(DateTypeQueryTest, FunctionTest) {
-  assertQuery("SELECT extract(year from col_b) FROM test", "functions/date/year.json");
-  assertQuery("SELECT extract(quarter from col_b) FROM test",
-              "functions/date/quarter.json");
-  assertQuery("SELECT extract(month from col_b) FROM test", "functions/date/month.json");
-  assertQuery("SELECT extract(day from col_b) FROM test", "functions/date/day.json");
-  assertQuery("SELECT extract(dayofweek from col_b) FROM test",
-              "functions/date/day_of_week.json");
-  assertQuery("SELECT extract(doy from col_b) FROM test",
-              "functions/date/day_of_year.json");
+  assertQueryArrow("SELECT extract(year from col_b) FROM test",
```

Comment:
I think we should keep origin test cases rather than replace it.

Reply:
doc updated,  and I found it difficult to maintain two sets of codepaths due to the new date type definition, constant/column  value representation, different type storage sizes, so migrated and fixed most  time related test cases in this pr (except string to date func).